synapse/rest/client/v1/login.py

```diff
@@ -144,99 +141,49 @@ async def _do_other_login(self, login_submission):
         # Convert deprecated authdict formats to the current scheme
         client_dict_convert_legacy_fields_to_identifier(login_submission)
 
-        if "identifier" not in login_submission:
-            raise SynapseError(400, "Missing param: identifier")
-
-        identifier = login_submission["identifier"]
-        if "type" not in identifier:
-            raise SynapseError(400, "Login identifier has no type")
-
-        # convert phone type identifiers to generic threepids
-        if identifier["type"] == "m.id.phone":
-            identifier = login_id_phone_to_thirdparty(identifier)
-
-        # convert threepid identifiers to user IDs
-        if identifier["type"] == "m.id.thirdparty":
-            address = identifier.get("address")
-            medium = identifier.get("medium")
-
-            if medium is None or address is None:
-                raise SynapseError(400, "Invalid thirdparty identifier")
-
-            if medium == "email":
-                # For emails, transform the address to lowercase.
-                # We store all email addreses as lowercase in the DB.
-                # (See add_threepid in synapse/handlers/auth.py)
-                address = address.lower()
-
-            # We also apply account rate limiting using the 3PID as a key, as
-            # otherwise using 3PID bypasses the ratelimiting based on user ID.
```

Comment:
This comment was helpful, I think.

Reply:
Hmm, I think calling out the 3PID is being used as a key here as opposed to user_id elsewhere is useful, but the second half is sort of implied in the new code I think?

Reply:
I think this code is confusing enough that it is not bad to be explicit.